python/semantic_kernel/contents/image_content.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import base64
+import logging
+import mimetypes
+import sys
+from typing import Any, Literal, TypeVar
+
+from semantic_kernel.exceptions.content_exceptions import ContentInitializationError
+
+if sys.version < "3.11":
+    from typing_extensions import Self
+else:
+    from typing import Self
+
+from xml.etree.ElementTree import Element  # nosec
+
+from pydantic import Field, model_validator
+from pydantic_core import Url
+
+from semantic_kernel.contents.const import IMAGE_CONTENT_TAG, ContentTypes
+from semantic_kernel.contents.kernel_content import KernelContent
+
+logger = logging.getLogger(__name__)
+
+_T = TypeVar("_T", bound="ImageContent")
+
+
+class ImageContent(KernelContent):
+    """This represent image content.
+
+    This can be created either with a uri for a image or with the bytes data of the image.
+    Use the .from_image_file method to create an instance from a image file.
+        This reads the file and guesses the mime_type.
+    If both uri and data is provided, data will be used and a warning is logged.
+
+    Args:
+        inner_content (Any): The inner content of the response,
+            this should hold all the information from the response so even
+            when not creating a subclass a developer can leverage the full thing.
+        ai_model_id (str | None): The id of the AI model that generated this response.
+        metadata (dict[str, Any]): Any metadata that should be attached to the response.
+        uri (Url | None): The uri of the image.
+        data (bytes | None): The data of the image.
+        mime_type (str | None): The mime type of the image, only used with data.
+
+    Methods:
+        from_image_path: Create an instance from an image file.
+        __str__: Returns the string representation of the image.
+
+    Raises:
+        ValidationError: If neither uri or data is provided.
+
+    """
+
+    content_type: Literal[ContentTypes.IMAGE_CONTENT] = Field(IMAGE_CONTENT_TAG, init=False)  # type: ignore
+    uri: Url | None = None
+    data: bytes | None = None
+    mime_type: str | None = None
+
+    @model_validator(mode="after")
+    def validate_uri_and_or_data(self) -> Self:
+        """Validate that either uri or data is provided."""
+        if not self.uri and not self.data:
+            raise ContentInitializationError("Either uri or data must be provided.")
+        if self.uri and self.data:
+            logger.warning('Both "uri" and "data" are provided, "data" will be used.')
+        return self
+
+    def __str__(self) -> str:
+        """Return the string representation of the image."""
+        return (
+            f"data:{self.mime_type};base64,{ base64.b64encode(self.data).decode('utf-8')}"
```

Comment:
nit: additional white space before `base64.b64encode`. It should be:
```python
f"data:{self.mime_type};base64,{base64.b64encode(self.data).decode('utf-8')}"
```

Reply:
I'll check, openai isn't picky about it 😏